### ISSUE TITLE
[Bug]: close event listener arguments missing in v25

### ISSUE THREAD

**Issue by SimonSiefke** [electron/electron#38545]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

25.0.1

### What operating system are you using?

Ubuntu

### Operating System Version

Ubuntu 23.04

### What arch are you using?

x64

### Last Known Working Electron version

24.1.2

### Expected Behavior

event is an instance of Event

### Actual Behavior

event is an empty object

### Testcase Gist URL

https://gist.github.com/SimonSiefke/141a391c0b5677550832247ed3fce388

### Additional Information

Using this code 

```js
const handleClose = event=>{
  console.log(event)
}

function createWindow () {
  // Create the browser window.
  const mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      preload: path.join(__dirname, 'preload.js')
    }
  })

  mainWindow.on('close', handleClose )

  // and load the index.html of the app.
  mainWindow.loadFile('index.html')

  // Open the DevTools.
  // mainWindow.webContents.openDevTools()
}
```

This is the output in electron 24:
```
Electron v24.1.2 started.
{
  preventDefault: [Function: preventDefault],
  sender: BrowserWindow {
    setBounds: [Function (anonymous)],
    _events: [Object: null prototype] {
      blur: [Function (anonymous)],
      focus: [Function (anonymous)],
      show: [Function: visibilityChanged],
      hide: [Function: visibilityChanged],
      minimize: [Function: visibilityChanged],
      maximize: [Function: visibilityChanged],
      restore: [Function: visibilityChanged],
      close: [Function: handleClose]
    },
    _eventsCount: 8,
    devToolsWebContents: [Getter]
  }
}
```

and in electron Electron 25:

```
Electron v25.0.1 started.
{}
```

**Comment by dsanders11** [comment]
First repros in v25.0.0-nightly.20230214.

**Comment by george-thomas-hill** [comment]
This is affecting me, too.

**Comment by jkleinsc** [comment]
#37099 refactored how events are emitted.  @SimonSiefke is there something in particular you are looking for/depending on from the event object?

**Comment by george-thomas-hill** [comment]
In my case, my on "close" handler needs to be able to get the ID of the window that is being closed.

The way that I do that is like this:

```
            // Prepare to close gracefully.
            mainWindow.on("close", handleCloseWindow);
            function handleCloseWindow(event) {
                // Unsubscribe.
                const theBrowserWindow = event.sender;
                const theWebContents = theBrowserWindow.webContents;
                const theId = theWebContents.id;
                . . .
            }
```

It occurs to me that my handler function might be able to get the ID from the mainWindow constant, but I can't try that out right this minute.

**Comment by MarshallOfSound** [comment]
> It occurs to me that my handler function might be able to get the ID from the mainWindow constant, but I can't try that out right this minute.

This is how you're supposed to do it 👍 `event.sender` was never part of the official API for anything other than IPC events and it's existence was an unhappy side-effect of code reuse.

TLDR: This is an intentional change and any previous usage of `event.sender` should be easily handleable by instead referencing the original window you added the event listener on.
